### PR TITLE
No longer return error from poweroff when VM is already powering down

### LINKED ISSUE #3718: You cannot remove a running container. Stop the container before attempting removal or use -f
This problem came up once before, but there are evidently other paths that lead to this issue, so I'm opening a new ticket for it.

I have 8 log bundles exhibiting this behavior. The basic gist is something like this (from https://ci.vmware.run/vmware/vic/7580):

```
time=2017-01-18T23:58:18.940466263Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerRm:407] 933e252ce6fd 
time=2017-01-18T23:58:18.940486838Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).Stop:463] 933e252ce6fd063388d571849dc52f044a892c67becf175dfbcb63b6c1a5da52 
time=2017-01-18T23:58:19.009205746Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).IsRunning:540] 
time=2017-01-18T23:58:19.064936239Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).Get:136] 
time=2017-01-18T23:58:19.064968092Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).getImageByDigest:176] 
time=2017-01-18T23:58:19.064976007Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).getImageByDigest:176] [5.365µs]  
time=2017-01-18T23:58:19.064982277Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).Get:136] [52.179µs]  
time="2017-01-18T23:58:19Z" level=debug msg="Docker inspect - network settings = &types.NetworkSettings{NetworkSettingsBase:types.NetworkSettingsBase{Bridge:\"\", SandboxID:\"\", HairpinMode:false, LinkLocalIPv6Address:\"\", LinkLocalIPv6PrefixLen:0, Ports:nat.PortMap{}, SandboxKey:\"\", SecondaryIPAddresses:[]network.Address(nil), SecondaryIPv6Addresses:[]network.Address(nil)}, DefaultNetworkSettings:types.DefaultNetworkSettings{EndpointID:\"\", Gateway:\"\", GlobalIPv6Address:\"\", GlobalIPv6PrefixLen:0, IPAddress:\"\", IPPrefixLen:0, IPv6Gateway:\"\", MacAddress:\"\"}, Networks:map[string]*network.EndpointSettings{\"bridge\":(*network.EndpointSettings)(0xc420bf8000)}}" 
time=2017-01-18T23:58:19.065072283Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).IsRunning:540] [55.870906ms]  
time="2017-01-18T23:58:19Z" level=debug msg="UnmapPorts: map[]" 
time=2017-01-18T23:58:19.065728884Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).CommitContainerHandle:392] e0b73bc519a8d429a74366a0035ca0ad 
time=2017-01-18T23:58:19.915943186Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).CommitContainerHandle:392] [850.207033ms] e0b73bc519a8d429a74366a0035ca0ad 
time=2017-01-18T23:58:19.915969358Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).Stop:463] [975.479794ms] 933e252ce6fd063388d571849dc52f044a892c67becf175dfbcb63b6c1a5da52 
time=2017-01-18T23:58:19.930885801Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerRm:407] [990.415027ms] 933e252ce6fd 
time="2017-01-18T23:58:19Z" level=error msg="Handler for DELETE /v1.21/containers/933e252ce6fd returned error: You cannot remove a running container. Stop the container before attempting removal or use -f" 
```

### PR DESCRIPTION
**Pull request by jzt**
This change prevents `stop` from returning an error during a power off retry on an already stopping container. This fixes a small edge case where stop fails and the powered/powering off container state gets reset to `Running`, preventing `ContainerRm` from removing a container, even if `-f` is supplied. This also fixes a test that erroneously creates `/bin/top` containers without supplying `-t` (related: #3728).

Fixes #3718 

Requesting feedback on this approach from @sflxn, @hmahmood, @anchal-agrawal

### CODE REVIEW

**Comment by caglar10ur** [comment]
Is this the B side of https://github.com/vmware/vic/pull/3689?

File: lib/portlayer/exec/base.go [hunk 1]
```diff
@@ -300,9 +300,15 @@ func (c *containerBase) poweroff(ctx context.Context) error {
 			case *types.GenericVmConfigFault:
 
 				// Check if the poweroff task was canceled due to a concurrent guest shutdown
-				if len(terr.FaultMessage) > 0 && terr.FaultMessage[0].Key == vmNotSuspendedKey {
-					log.Infof("power off %s task skipped due to guest shutdown", c.ExecConfig.ID)
-					return nil
+				if len(terr.FaultMessage) > 0 {
+					switch terr.FaultMessage[0].Key {
+					case vmNotSuspendedKey:
+						fallthrough
```
Comment (anchal-agrawal) [inline lib/portlayer/exec/base.go]: If we intend to do the same thing for both these cases, we can say `case vmNotSuspendedKey, vmPoweringOffKey:` instead of `fallthrough`. Talked about it with @matthewavery too, he also agrees the former is more readable.

**Comment by jzt** [comment]
@caglar10ur definitely, but #3689 does not address the root cause of #3718.

File: lib/portlayer/exec/base.go [hunk 2]
```diff
@@ -302,9 +302,13 @@ func (c *containerBase) poweroff(ctx context.Context) error {
 			case *types.GenericVmConfigFault:
 
 				// Check if the poweroff task was canceled due to a concurrent guest shutdown
-				if len(terr.FaultMessage) > 0 && terr.FaultMessage[0].Key == vmNotSuspendedKey {
-					log.Infof("power off %s task skipped due to guest shutdown", c.ExecConfig.ID)
-					return nil
+				if len(terr.FaultMessage) > 0 {
+					switch terr.FaultMessage[0].Key {
```
Comment (vburenin) [inline lib/portlayer/exec/base.go]: I think using switch is an overkill:

```
k := terr.FaultMessage[0].Key
// These faults are no longer relevant during shutdown.
if k == vmNotSuspendedKey || k == vmPoweringOffKey {
    log....
    return nil
}
```

Comment (jzt) [inline lib/portlayer/exec/base.go]: 👍

**Comment by mhagen-vmware** [comment]
lgtm

**Comment by sflxn** [comment]
lgtm
